clients/mobile/src/myActivity/donorProfile/useDonarProfile.ts

```diff
@@ -0,0 +1,44 @@
+import { useRoute } from '@react-navigation/native'
+import { DonarProfileRouteProp } from '../../setup/navigation/navigationTypes'
+import { useEffect, useState } from 'react'
+import { useFetchClient } from '../../setup/clients/useFetchClient'
+import { Alert, Linking } from 'react-native'
+import { getDonarProfile, DonorProfile } from '../../userWorkflow/services/userServices'
+
+const useDonarProfile = (): any => {
+  const fetchClient = useFetchClient()
+  const { donarId } = useRoute<DonarProfileRouteProp>().params
+  const [donarProfile, setDonarProfile] = useState<DonorProfile | null>(null)
+  const [loading, setLoading] = useState<boolean>(true)
+  const [error, setError] = useState<string | null>(null)
+
+  useEffect(() => { void getDonarInfo() }, [donarId])
+
+  const getDonarInfo = async(): Promise<void> => {
+    try {
+      const response = await getDonarProfile(donarId, fetchClient)
+      if (response.data !== null && response.data !== undefined) {
+        setDonarProfile(response.data)
+      }
+    } catch (err) {
+      setError('Failed to fetch donor profile.')
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  const handleCall = (): void => {
+    if (donarProfile !== null && Array.isArray(donarProfile.phoneNumbers) && donarProfile.phoneNumbers.length > 0) {
+      const phoneNumber = donarProfile.phoneNumbers[0]
+      Linking.openURL(`tel:${phoneNumber}`).catch(() => {
+        Alert.alert('Error', 'Unable to make a call. Please try again later.')
+      })
+    } else {
```

Comment:
use a **_guard clause_**, which allows you to exit early if a condition is not met

```
const handleCall = (): void => {
  if (donarProfile === null || !Array.isArray(donarProfile.phoneNumbers) || donarProfile.phoneNumbers.length === 0) {
    Alert.alert('No Phone Number', 'No phone number available for this donor.');
    return;
  }

  const phoneNumber = donarProfile.phoneNumbers[0];
  Linking.openURL(`tel:${phoneNumber}`).catch(() => {
    Alert.alert('Error', 'Unable to make a call. Please try again later.');
  });
};
```

Reply:
Is there any downside to writing an `else` here? It's just a simple `if-else`; it's not like we're doing too much before writing the `else`.

Reply:
It's a good practice to avoid branching code whenever you can. It makes the code more readable.

Reply:
and cognitive load, makes it harder for reviewers to read

Reply:
Updated [Commit](https://github.com/CraftsmenLtd/BloodConnect/pull/295/commits/14e368f3cf7b2d686f72f9c468cbe55f9a276229)